### PR TITLE
[docs] Spanish translation of model_memory_anatomy.md

### PR DESCRIPTION
**Pull request by aaronjimv**
# What does this PR do?

Add the Spanish version of `model_memory_anatomy.md` to `transformers/docs/source/es`

#28936

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@stevhliu

### CODE REVIEW

**Comment by aaronjimv** [comment]
cc: @tadeodonegana , @gisturiz

Hello guys, I would appreciate if you could review the translation, it is a very technical documentation so I am open to any feedback. Thank you for your help.

File: docs/source/es/model_memory_anatomy.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+```
+
+Comencemos comprobando que la memoria GPU este libre:
+
+```py
+>>> print_gpu_utilization()
+GPU memory occupied: 0 MB.
+```
+
+Parece estar bien: la memoria de la GPU no está ocupada como esperaríamos antes de cargar cualquier modelo. Si no es el caso en tu máquina, asegúrate de detener todos los procesos que estén utilizando la memoria de la GPU. Sin embargo, no toda la memoria libre de la GPU puede ser utilizada por el usuario. Cuando se carga un modelo en la GPU, también se cargan los kernels, lo que puede ocupar 1-2GB de memoria. Para ver cuánta es, cargemos un tensor diminuto en la GPU, lo que también desencadena la carga de los kernels.
```
Comment (tadeodonegana) [inline docs/source/es/model_memory_anatomy.md:72]: I suggest changing:

_"Para ver cuánta es"_

to

_"Para ver cuánta memoria será ocupada por defecto"_,

to clarify that we are referring to the memory that will be taken by kernels and to make that more clear.

File: docs/source/es/model_memory_anatomy.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+<Tip>
+
+Si planeas ejecutar varias pruebas, reinicie el kernel de Python entre cada prueba para borrar correctamente la memoria.
+
+</Tip>
+
+## Utilización de la memoria en el entrenamiento
+
+Vamos a utilizar el [`Trainer`] y entrenar el modelo sin utilizar ninguna técnica de optimización del rendimiento de la GPU y un tamaño de lote de 4:
```
Comment (tadeodonegana) [inline docs/source/es/model_memory_anatomy.md:147]: I suggest referencing the [Trainer](https://huggingface.co/docs/transformers/v4.41.0/en/main_classes/trainer#transformers.Trainer) docs page, as done in the original page.

**Review by tadeodonegana — COMMENTED** [review]
Amazing work as always, @aaronjimv! I just left some minimal comments.

Thanks for your work 🥇

File: docs/source/es/model_memory_anatomy.md (hunk 1 above)
Comment (aaronjimv) [reply docs/source/es/model_memory_anatomy.md:72]: Thanks for your feedback! I really appreciate it.

File: docs/source/es/model_memory_anatomy.md (hunk 2 above)
Comment (aaronjimv) [reply docs/source/es/model_memory_anatomy.md:147]: Ok, that seems fine to me. Thanks.

**Review by tadeodonegana — APPROVED** [review]
LGTM

**Comment by aaronjimv** [comment]
> LGTM

Thanks!

Hello @stevhliu, let me know anything about the changes. Thanks🤗

**Review by stevhliu — APPROVED** [review]
Nice work! 👏

File: docs/source/es/model_memory_anatomy.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+>>> seq_len, dataset_size = 512, 512
+>>> dummy_data = {
+...     "input_ids": np.random.randint(100, 30000, (dataset_size, seq_len)),
+...     "labels": np.random.randint(0, 1, (dataset_size)),
+... }
+>>> ds = Dataset.from_dict(dummy_data)
+>>> ds.set_format("pt")
+```
+
+Para imprimir estadísticas resumidas para la utilización de la GPU y la ejecución del entrenamiento con [`Trainer`](https://huggingface.co/docs/transformers/v4.41.0/en/main_classes/trainer#transformers.Trainer), definimos dos funciones auxiliares:
```
[the remark this answers is not in the document]
Comment (aaronjimv) [reply docs/source/es/model_memory_anatomy.md:46]: Ok.

**Review by stevhliu — APPROVED** [review]
Thanks!

**Pull request merged**